WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -15,45 +15,48 @@ public partial class LabelSelectionViewModel : ViewModelBase
 
 	[AutoNotify] private bool _enoughSelected;
 
-	private Pocket? _privatePocket;
+	private Pocket _privatePocket = Pocket.Empty;
 	private bool _includePrivatePocket;
+	private Pocket[] _allPockets = Array.Empty<Pocket>();
 
 	public LabelSelectionViewModel(Money targetAmount)
 	{
 		_targetAmount = targetAmount;
 	}
 
-	public Pocket[] AllPockets { get; set; } = Array.Empty<Pocket>();
+	public Pocket[] NonPrivatePockets { get; set; } = Array.Empty<Pocket>();
 
 	public IEnumerable<LabelViewModel> AllLabelsViewModel { get; set; } = Array.Empty<LabelViewModel>();
 
 	public IEnumerable<LabelViewModel> LabelsWhiteList => AllLabelsViewModel.Where(x => !x.IsBlackListed);
 
 	public IEnumerable<LabelViewModel> LabelsBlackList => AllLabelsViewModel.Where(x => x.IsBlackListed);
 
-	public Pocket[] GetSafeToUsePockets()
+	public Pocket[] AutoSelectPockets()
 	{
-		var pockets = AllPockets.ToList();
+		var knownPockets = NonPrivatePockets.Where(x => x.Labels != CoinPocketHelper.UnlabelledFundsText).ToArray();
 
-		if (_privatePocket is { } privatePocket)
+		if (_privatePocket.Amount >= _targetAmount)
 		{
-			pockets.Add(privatePocket);
+			return new[] { _privatePocket };
 		}
 
-		var unlabelledPocket = pockets.FirstOrDefault(x => x.Labels == CoinPocketHelper.UnlabelledFundsText);
+		if (knownPockets.Sum(x => x.Amount) >= _targetAmount)
+		{
+			return knownPockets;
+		}
 
-		if (unlabelledPocket is { } && pockets.Where(x => x != unlabelledPocket).Sum(x => x.Amount) >= _targetAmount)
+		if (NonPrivatePockets.Sum(x => x.Amount) >= _targetAmount)
```

Comment:
Unknown pocket is the worst to use. We have no information about that. That is why it is only included when absolutely necessary when the user wouldn't be able to make the transaction without that.

Reply:
Doesn't `NonPrivatePockets` include `unknownPockets`?

If yes and if unknown pocket is the worst to use then here we are using unknown pocket when we might be able to use the private pocket + the known pocket.

Reply:
Sorry, I was not clear. It is almost the worst. Mixing non private & private is the worst.

This is PR is fixing that case that you mentioned. Because right now on the master, it rather mixes the non private & private funds just to avoid the unknown pockets.

Reply:
In this PR... it tries to use first the private funds, if not enough then the known funds, if not enough the known + unknown funds, if not enough the non private funds + private funds (which is the total balance).

Reply:
In this PR... it tries to use first the private funds, if not enough then the known funds, if not enough the known + unknown funds, if not enough the non private funds + private funds (which is the total balance).

Reply:
AFAIU mixing/merging pockets should be the last so as you said mixing the known + unknown funds, and the non private funds + private funds should be the last options.

But I think using the unknown funds alone if possible is better than using the known + unknown funds, because the unknown observer/entity would not learn about your known funds.

Reply:
As far as I understand there are 3 categories of pockets: private, known, unknown.
This is how they can be used if they cover the target amount:

1. private
2. known
3. unknown (this is missing)
4. known + unknown
5. private + known (this is also missing)
6. private + unknown (this is also missing)
7. all 3 pockets

Reply:
Fixed the missing ones. I added tests as well.

Reply:
Very good job